hazelcast/src/main/java/com/hazelcast/spi/impl/merge/AbstractSplitBrainHandlerService.java

```diff
@@ -135,24 +128,39 @@ public int getPartitionId() {
     }
 
     private boolean isLocalPartition(int partitionId) {
-        IPartition partition = partitionService.getPartition(partitionId, false);
-        return partition.isLocal();
+        return partitionService.isPartitionOwner(partitionId);
     }
 
-    // overridden in implementations
-    protected boolean hasDiscardPolicy(Object mergePolicy) {
-        return mergePolicy instanceof DiscardMergePolicy;
-    }
+    /**
+     * Returns a runnable which merges the given {@link Store} instances.
+     *
+     * @return a merge runnable for the given stores
+     */
+    protected abstract Runnable newMergeRunnable(Collection<Store> mergingStores);
 
-    protected abstract void destroyStore(Store store);
+    /**
+     * Returns an {@link Iterator} over all {@link Store} instances in the given partition.
+     *
+     * @return an iterator over all stores
+     */
+    protected abstract Iterator<Store> storeIterator(int partitionId);
 
-    protected abstract boolean hasEntry(Store store);
+    /**
+     * Destroys the given {@link Store}.
+     */
+    protected abstract void destroyStore(Store store);
 
-    protected abstract Runnable newMergeRunnable(Collection<Store> mergingStores,
-                                                 BaseSplitBrainHandlerService<Store> splitBrainHandlerService);
```

Comment:
An abstract class is not allowed to be used as parameter (CheckStyle doesn't like this). This just worked, because the class was named `BaseSplitBrainHandlerService`.

Luckily we can just pass `this` in the implementations, no need to pass the service to itself.